open-company/open-company-web
# REFACTOR: Move UI related actions

Card: https://trello.com/c/nOF8CduC

Actions:
```
:error-banner-show
:whats-new/finish
```

The error banner action is moved to the error-banner action creator and store since it's called from different places of the code. Move it to the component like i did for some of the following action wasn't very clean since i didn't like including a component namespace into `oc.web.api` for example.
Whats new load and loaded action is moved in the user action creator since that's where it start loading.

Removed actions:
```
:made-with-carrot-modal-show
:made-with-carrot-modal-hide
:org-settings-show
:org-settings-hide
:whats-new-modal-show
:whats-new-modal-hide
:site-menu-toggle
:mobile-menu-toggle
:trend-bar-status // completely removed
:alert-modal-show
:alert-modal-hide
:alert-modal-hide-done
```

These actions have been removed completely since are all actions related to a component that is shown only one time in the page so we can use a function inside the component namespace that set the needed data into the app-state to show the component and remove it when done.


### Tests

### Alert modal

User when deleting a comment, deleting a draft, dismissing post create w/ unsaved changes or while uploading an attachment or image, dismiss post update w/ unsaved changes or while uploading an attachment or image, exit org settings with unsaved edits, error while uploading an image or attachment in the post edit, dismiss user profile with unsaved changes, delete a post, an error occurred while uploading a new company avatar, resend an invite, remove a user or cancel an invitation, delete a section.

Test 1:
- go to draft
- click delete
- [x] do you see the prompt view? Good
- [x] can you dismiss without removing it? Good
- [x] can you remove it? Good

Test 2:
- go to create a new post
- add a title and a body
- click on the X in top left corner
- [x] do you see the prompt to leave losing changes? Good
- [x] can you click cancel and stay there? Good
- [x] can you click ok and lose the changes? Good

Test 3:
- edit a post
- edit headline and body
- click X in top left corner
- [x] do you see the prompt? Good
- [x] can you click cancel and stay there? Good
- [x] can you click ok and lose the changes? Good

Test 4:
- click the user menu
- go to Team Settings
- edit the company name
- click the X in top left corner
- [x] do you see the prompt? Good
- [x] can you click cancel and stay there? Good
- [x] can you click ok and lose the changes? Good

Test 5:
- click the user menu
- go to User Profile
- change your first and last name
- click closing X
- [x] do you see the prompt? Good
- [x] can you click cancel and stay there? Good
- [x] can you click ok and lose the changes? Good

Test 6:
- go to AP stream view
- click the vertical ellipses
- click Delete
- [x] do you see the prompt? Good
- [x] can you click cancel and stay there? Good
- [x] can you click ok and remove the post? Good
- [x] repeat in grid view

Test 7:
- click user menu
- click Team Management
- if you don't have already a pending invited user, invite one via email
- click on Resend invite
- [x] do you see the prompt of invite resent? Good
- click on the role dropdown for an active user or a pending user
- click on Remove
- [x] do you see the prompt to remove the user? Good
- [x] can you click cancel and stay there? Good
- [x] can you click ok and remove the user? Good

Test 8:
- go into a section
- click on Section settings
- click Delete section at the bottom
- [x] do you see the prompt to confirm delete? Good
- [x] can you click cancel and stay there? Good
- [x] can you click ok and remove the section? Good

#### Testing :mobile-menu-toggle

Test:
- visit org dashboard on mobile
- click on the user avatar in top right corner
- [x] does the user menu expand? Good
- [x] can you dismiss it clicking again on the avatar? Good

#### Testing :site-menu-toggle

Test:
- visit /login in incognito on mobile
- click the X to dismiss the login view and go back to the homepage
- click on the ham menu in top left corner
- [x] does the site menu appear? Good
- now click on the X in top left corner
- [x] does the site menu go away? Good

#### Testing What's New modal

Test:
- go to staging.carrot.io
- go to your digest or login with slack/email
- check the console
- [x] do you see the `:whats-new/finish` action? Good
- now click on Support in bottom left of the page
- [x] do you see the what's new modal appear? Good
- [ ] do you see some post if the what's new load wasn't a 404? Good

#### Testing Org Settings show/hide

Test:
- click on the user menu
- click on Team Settings
- [x] do you see the team settings appear? Good
- [x] can you switch to invite and team management? Good
- [x] can you dismiss the settings? Good
- click on the user menu again
- click on Team Management
- [x] do you see the Team Management settings panel appear? Good
- dismiss the settings panel
- click on the user menu
- click on Invite People
- [x] do you see the invite people panel appear? Good
- dismiss it
- click on Invite People in bottom left corner of the page
- [x] do you see the invite people panel appear? Good
- dismiss it

#### Testing Made With Carrot show/hide

Test:
- click on the vertical ellipses of a post
- click share
- click URL
- copy the URL
- in an incognito window paste the url and go
- [x] do you see the secure activity page? Good
- click on made with Carrot at the bototm
- [x] do you see the Made With Carrot modal appear? Good
- [x] can you dismiss it? Good

#### Testing Error Banner

Test:
- visit your dashboard
- stop the storage service
- go back to the dashboard and switch board or to AP
- [x] do you see the error banner appear? Good
- [x] does it dismiss after 10 seconds? Good
